newlib/libc/sys/xtensa/sys/string.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * sys/string.h
+ *
+ * Xtensa custom PROGMEM string function definitions
+ */
+
+#ifndef _SYS_STRING_H_
+#define	_SYS_STRING_H_
+
+#include "_ansi.h"
+#include <sys/reent.h>
+#include <sys/cdefs.h>
+#include <sys/features.h>
+
+#define __need_size_t
+#define __need_NULL
+#include <stddef.h>
+
+#define SIZE_IRRELEVANT 0x7fffffff
```

Comment:
Could this be changed to a static const or enum?

Reply:
I can see an enum (static const is an extra 4 bytes of heap required + an extra 2-3 insns on the routines to load its address and then load the value) ore prepending a __ to the macro to identify it as internal.

Any reason why, though?  This was just pulled direct from the existing code. If I was to really clean up technical debt, the proper thing would be to move this enum/macro to a separate namespace, or replce the #defines w/real wrapper functions where the SIZE_IRRELEVANT is hidden completely from the outside world.  There is no userland reason to ever employ it.